data/org.freedesktop.portal.ScreenCast.xml

```diff
@@ -0,0 +1,182 @@
+        using org.freedesktop.portal.ScreenCast::SelectSources.
+
+        The @parent_window identifier must be of the form "x11:$XID" for an X11
+        window. Support for other window systems may be added in the future.
+
+        The following results get returned via the
+        #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>streams a(ua{sv})</term>
+            <listitem><para>
+              An array of PipeWire streams. Each stream consists of a PipeWire
+              node ID (the first element in the tuple, and a Vardict of
+              properties.
+
+              The array will contain a single stream if 'multiple' (see
+              SelectSources) was set to 'false', or at least one stream if
+              'multiple' was set to 'true' as part of the SelectSources method.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+
+        Stream properties include:
+        <variablelist>
+          <varlistentry>
+            <term>position (ii)</term>
+            <listitem><para>
+              A tuple consisting of the position (x, y) in the compositor
+              coordinate space. Note that the position may not be equivalent to a
+              position in a pixel coordinate space. Only available for monitor
+              streams.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>size (ii)</term>
+            <listitem><para>
+              A tuple consisting of (width, height). The size represents the size
+              of the stream as it is displayed in the compositor coordinate
+              space. Note that this size may not be equivalent to a size in a
+              pixel coordinate space. The size may differ from the size of the
+              stream.
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="Start">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="parent_window" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
+    <!--
+        TakePipeWireRemote:
+        @session_handle: Object path for the #org.freedesktop.portal.Session object
+        @options: Vardict with optional further information
+        @fd: File descriptor of an open PipeWire remote.
+
+        Retrieve the file descriptor to the PipeWire remote where the screen
+        cast streams are available. The file descriptor should be used to
+        create a <classname>pw_remote</classname> object, by using
+        <function>pw_remote_connect_fd</function>.
+
+        This function can only be called once, if there was one or more stream
+        shared in the response to starting the session.
+    -->
+    <method name="TakePipeWireRemote">
```

Comment:
I think you need:
```
      <annotation name="org.gtk.GDBus.C.UnixFD" value="true"/> 
```
here, so that you get the right codegen for the fds.

Reply:
Oh, i see you did that in the makefile. We're changing those to inline annotations instead.